sklearn/decomposition/_factor_analysis.py

```diff
@@ -204,6 +200,13 @@ def fit(self, X, y=None):
         self : object
             FactorAnalysis class instance.
         """
+
+        if self.svd_method not in ["lapack", "randomized"]:
+            raise ValueError(
+                "SVD method %s is not supported. Please consider the documentation"
+                % self.svd_method
```

Comment:
Since we are here, I think we can update to using f-strings here:

```suggestion
                f"SVD method {self.svd_method!r} is not supported. Please consider the documentation"
```